src/coreclr/tools/aot/ILCompiler.TypeSystem.Tests/TypeEquivalenceAssembly/TypesWhichDoNotLoad.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using System.Runtime.InteropServices;
+
+#pragma warning disable 169
+
+namespace TypesWhichDoNotLoad
+{
+    // A case where a field is not public
+    [TypeIdentifier("TypesWhichDoNotMatch", "Type1")]
+    public struct Type1
+    {
+        int field;
+    }
+
+    // A case where the type is not both public
```

Comment:
Nit - I'm missing the purpose of the word *both* in this code comment (unless *both public* is a technical term I haven't yet come across); dtto for the next two test cases.